Monika After Story/game/script-songs.rpy

```diff
@@ -59,6 +68,53 @@ init python in mas_songs:
             if ev.random and TYPE_SHORT in ev.category and ev.checkAffection(store.mas_curr_affection)
         ]
 
+    def checkSongAnalysisDelegate(curr_aff=None):
+        """
+        Checks to see if the song analysis topic should be unlocked or locked and does the appropriate action
+
+        IN:
+            curr_aff - Affection level to ev.checkAffection with. If none, mas_curr_affection is assumed
+                (Default: None)
+        """
+        if hasUnlockedSongAnalyses(curr_aff):
+            store.mas_unlockEVL("monika_sing_song_analysis", "EVE")
+        else:
+            store.mas_lockEVL("monika_sing_song_analysis", "EVE")
+
+    def getUnlockedSongAnalyses(curr_aff=None):
+        """
+        Gets a list of all song analysis evs in scrollable menu format
+
+        IN:
+            curr_aff - Affection level to ev.checkAffection with. If none, mas_curr_affection is assumed
+                (Default: None)
+
+        OUT:
+            List of unlocked song analysis topics in mas_gen_scrollable_menu format
+        """
+        if curr_aff is None:
+            curr_aff = store.mas_curr_affection
+
+        return [
+            (ev.prompt, ev_label, False, False)
```

Comment:
So with this we'll have to have the same prompts for both song and its analysis? I'm not sure if it's ok, so I thought I'd let you know. I think we could:
* use something like `"<song_name> analysis"` (which sounds a bit dumb/formal?)
* or maybe fetch the song's prompt and use it in this menu while not specifying the prompt for the analysis' event at all?
* we also could do nothing if you think it's safe to use the same prompts.

Reply:
I think it's fine to use the same prompt. This is in a separate topic where you ask her to explain a song, in said topic, she asks which song you'd like her to explain. So using the actual song name works out